test/extended/images/helper.go

```diff
@@ -327,3 +332,79 @@ func calculateRoughDataSize(logger io.Writer, wantedImageSize uint64, numberOfLa
 	// running Docker daemon < 1.9
 	return uint64(float64(wantedImageSize) / (float64(numberOfLayers) * layerSizeMultiplierForDocker18))
 }
+
+// MirrorBlobInRegistry forces a blob of external image to be mirrored in the registry. The function expects
+// the blob not to exist before a GET request is issued. The function blocks until the blob is mirrored or the
+// given timeout passes.
+func MirrorBlobInRegistry(oc *exutil.CLI, dgst digest.Digest, repository string, timeout time.Duration) error {
+	presentGlobally, inRepository, err := IsBlobStoredInRegistry(oc, dgst, repository)
+	if err != nil {
+		return err
+	}
+	if presentGlobally || inRepository {
+		return fmt.Errorf("blob %q is already present in the registry", dgst.String())
+	}
+	registryURL, err := GetDockerRegistryURL(oc)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s/v2/%s/blobs/%s", registryURL, repository, dgst.String()), nil)
+	if err != nil {
+		return err
+	}
+	token, err := oc.Run("whoami").Args("-t").Output()
+	if err != nil {
+		return err
+	}
+	req.Header.Set("range", "bytes=0-1")
+	req.Header.Set("Authorization", "Bearer "+token)
+	c := http.Client{}
+	resp, err := c.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusPartialContent && resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("expected status %d for request '%s %s', got %d", http.StatusOK, req.Method, req.URL.String(), resp.StatusCode)
+	}
+
+	return wait.Poll(time.Second, time.Minute, func() (bool, error) {
+		globally, inRepo, err := IsBlobStoredInRegistry(oc, dgst, repository)
+		return globally || inRepo, err
+	})
+}
+
+// IsEmptyDigest returns true if the given digest matches one of empty blobs.
+func IsEmptyDigest(dgst digest.Digest) bool {
+	return dgst == digestSha256EmptyTar || dgst == digestSHA256GzippedEmptyTar
+}
+
+// IsBlobStoredInRegistry verifies a presence of the given blob on registry's storage. The registry must be
+// deployed with a filesystem storage driver. If repository is given, the presence will be verified also for
+// layer link inside the ${repository}/_layers directory.
+func IsBlobStoredInRegistry(
+	oc *exutil.CLI,
+	dgst digest.Digest,
+	repository string,
+) (presentGlobally, presentInRepository bool, err error) {
```

Comment:
Drop named returns, they are confusing and you're using it only once at the end.